### PR TITLE
Change pool withdraw functions to signatures, update tests and tools

### PR DESCRIPTION
**Pull request by AboldUSER**
Updated the pool contract to EIP712, and switched all withdraw functions to signature validation. Removed all merkle validation functionality. Updated tests, and updated tools (utils, typescript, constants) to include claim signatures and differentiate from order signatures.

### CODE REVIEW

**Review by francisldn — COMMENTED** [review]
2 questions from me regarding the ``_markNonceAsUsed`` function and expiry argument

File: source/pool/contracts/Pool.sol [hunk 1]
```diff
@@ -365,37 +344,110 @@ contract Pool is IPool, Ownable {
...
+    require(_markNonceAsUsed(participant, nonce), "NONCE_ALREADY_USED");
+  }
+
+  /**
+   * @notice Marks a nonce as used for the given participant
+   * @param participant address
+   * @param nonce uint256
+   * @return bool True if nonce was not marked as used already
+   */
+  function _markNonceAsUsed(address participant, uint256 nonce)
```
[the remark this answers is not in the document]
Comment (AboldUSER) [reply source/pool/contracts/Pool.sol:444]: I've updated the function to use a bit operator pattern and added another public function to passed boolean validation if a nonce has been used, based on the swap contract.

File: source/pool/contracts/Pool.sol [hunk 2]
```diff
@@ -197,161 +190,147 @@ contract Pool is IPool, Ownable {
...
+   * @param nonce uint256
+   * @param score uint256
+   * @param v uint8 "v" value of the ECDSA signature
+   * @param r bytes32 "r" value of the ECDSA signature
+   * @param s bytes32 "s" value of the ECDSA signature
    */
-  function withdraw(Claim[] memory claims, address token) external override {
-    withdrawProtected(claims, token, 0, msg.sender);
+  function withdraw(
+    address token,
```
[the remark this answers is not in the document]
Comment (AboldUSER) [reply source/pool/contracts/Pool.sol:203]: We are now using a sequential nonce per participant. So an admin (e.g., Activate) could check if the previous nonce was used before issuing a new signature/nonce. Hence an expiry is not necessary for this implementation.

**Review by dmosites — APPROVED** [review]
LGTM

**Pull request merged**